exec/store_test.go

```diff
@@ -70,6 +70,26 @@ func testStore(t *testing.T, store Store) {
 	if !bytes.Equal(data, got) {
 		t.Error("data do not match")
 	}
+	// No partition 1 was ever stored, so discarding is a no-op.
+	err = store.Discard(ctx, task, 1)
+	if err != nil {
+		t.Error(err)
+	}
+	// Make sure we can still Open successfully after the unrelated Discard.
+	rc, err = store.Open(ctx, task, 0, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rc.Close()
+	// Now discard, and try to Open. It should fail.
```

Comment:
Okay; I clarified `Discard`'s contract to be more than best-effort.

Reply:
Now the `(Store).Discard` is more than best-effort, but the `(*Result).Discard` remains best-effort to hypothetically deal with transient errors that affect discarding but may not affect computation correctness. I think this is safe because I think that anything that would be affected by partial partition deletion would also be affected by recomputed tasks due to faults like lost machines.